apps/srt-live-transmit.cpp

```diff
@@ -771,12 +777,49 @@ int main(int argc, char** argv)
                     break;
                 }
 
+
+                bool srcReady = false;
+
+                if (src.get() && src->IsOpen())
+                {
+                    if (srtrfdslen > 0)
+                    {
+                        SRTSOCKET sock = src->GetSRTSocket();
+                        if (sock != SRT_INVALID_SOCK)
+                        {
+                            for (int n = 0; n < srtrfdslen; n ++)
+                            {
+                                if (sock == srtrwfds[n])
+                                {
+                                    srcReady = true;
+                                    break;
+                                }
+                            }
```

Comment:
Consider packing this into a helper function than can be used both for `srtrwfds` and `sysrfds`, thus making the code more compact and easier to read the intent.